compiler/vm/vm.nim

```diff
@@ -1990,100 +2743,100 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
 
 
     of opcCallSite:
-      ensureKind(rkNode)
-      if c.callsite != nil: regs[ra].node = c.callsite
-      else: stackTrace(c, tos, pc,  reportStr(rsemVmFieldNotFound, "callsite"))
+      ensureKind(rkNimNode)
+      if c.callsite != nil: regs[ra].nimNode = c.callsite
+      else: 
+        stackTrace(c, tos, pc,  reportStr(rsemVmFieldNotFound, "callsite"))
+        exitWithError()
     of opcNGetLineInfo:
-      decodeBImm(rkNode)
-      let n = regs[rb].node
+      decodeBImm()
+      let n = regs[rb].nimNode
       case imm
       of 0: # getFile
-        regs[ra].node = newStrNode(nkStrLit, toFullPath(c.config, n.info))
+        regs[ra].initValueReg(c.typeInfoCache.stringType, c.memory)
+        regs[ra].strVal.newVmString(toFullPath(c.config, n.info), c.memory)
       of 1: # getLine
-        regs[ra].node = newIntNode(nkIntLit, n.info.line.int)
+        regs[ra].initIntReg(n.info.line.int)
       of 2: # getColumn
-        regs[ra].node = newIntNode(nkIntLit, n.info.col)
+        regs[ra].initIntReg(n.info.col)
       else:
-        internalAssert(c.config, false, "Unexpected opcNGetLineInfo action code")
-
-      regs[ra].node.info = n.info
-      regs[ra].node.typ = n.typ
+        assert false # vmgen issue
+        #internalAssert(c.config, false, "Unexpected opcNGetLineInfo action code")
     of opcNSetLineInfo:
-      decodeB(rkNode)
-      regs[ra].node.info = regs[rb].node.info
+      decodeB(rkNimNode)
+      regs[ra].nimNode.info = regs[rb].nimNode.info
     of opcEqIdent:
       decodeBC(rkInt)
-      # aliases for shorter and easier to understand code below
-      var aNode = regs[rb].node
-      var bNode = regs[rc].node
-      # Skipping both, `nkPostfix` and `nkAccQuoted` for both
-      # arguments.  `nkPostfix` exists only to tag exported symbols
-      # and therefor it can be safely skipped. Nim has no postfix
-      # operator. `nkAccQuoted` is used to quote an identifier that
-      # wouldn't be allowed to use in an unquoted context.
-      if aNode.kind == nkPostfix:
-        aNode = aNode[1]
-      if aNode.kind == nkAccQuoted:
-        aNode = aNode[0]
-      if bNode.kind == nkPostfix:
-        bNode = bNode[1]
-      if bNode.kind == nkAccQuoted:
-        bNode = bNode[0]
+
+      func asCString(a: TFullReg): cstring =
```

Comment:
Doesn't need to be done now, but this feels like callback ("syscall") territory.